### DISCUSSION TITLE
Data availability if computer not running during a drive

### DISCUSSION THREAD

**Discussion by SteveDonie** [teslamate-org/teslamate#3209] [Q&A] [answered]
I have Teslamate running on my laptop, and it works great. What I don't understand is the 'steps' that data takes from my car to the application. Is the data first streamed to Tesla servers, or is it sent directly to my laptop? If it is sent to Tesla servers, do they keep it for any length of time? When Teslamate restarts, does it poll for any data since the last time it was 'online'? 

I noticed that a drive I took while my laptop was turned off was not recorded in the application, so it seems like if the data isn't received by the Teslamate application at the time it is sent, it just gets lost.

**Comment by cwanja** [comment, marked as answer]
Your last statement is correct. If TeslaMate is not running, it will not collect the data. Tesla likely preserves the data (for their own purpose), but it is not available to third-party applications to pull from. As such, if TeslaMate is down - you will never retrieve the data.

**Comment by SteveDonie** [comment]
Thanks, answer is appreciated.
